### PR TITLE
run all tests on GitHub Actions

### PR DESCRIPTION
**Pull request by ocefpaf**
Just a draft for now. I want to add the downstream tests too and then remove the .travis.yml file.

### CODE REVIEW

**Comment by ocefpaf** [comment]
This one is ready for review. The failures here are the same we have in current `master`.

File: pytest.ini [hunk 1]
```diff
@@ -1,5 +0,0 @@
-[pylama]
-format = pylint
-skip = */doc/*,*/data/*
-linters = pep8,pyflakes
-ignore = E501,E251,E221,E201,E202,E203
```
Comment (ocefpaf) [inline pytest.ini:5]: Need to migrate these to the pre-commit hooks. I'll  that in another PR and fix the pre-commits failures at the same time.

**Comment by ocefpaf** [comment]
The NCEI plugin was failing before and it seems to be related with the plugin code and not compliance-checker. BTW, the plugins tests where "hidden" in Travis-CI under the allow failure. I'm not sure we can do that with GitHub Actions but I believe it is better this way, so the failures are explicitly when we decide to merge.

@daltonkell and @benjwadams this PR is ready for review!

**Comment by benjwadams** [comment]
Looks good to me.

**Pull request merged**